### PR TITLE
Add StackView.squeeze() method, like numpy.squeeze()

### PR DESCRIPTION
**Pull request by takluyver**
Closes gh-15

### CODE REVIEW

File: extra_data/stacking.py [hunk 1]
```diff
@@ -208,3 +208,8 @@ def asarray(self):
         for modno, data in self._data.items():
             arr[modno] = data
         return np.moveaxis(arr, 0, self._stack_axis)
+
+    def squeeze(self):
+        """Drop any length-1 dimensions - see numpy.squeeze()"""
+        slices = [0 if d == 1 else slice(None, None) for d in self.shape]
```
Comment (zhujun98) [inline extra_data/stacking.py]: Thanks for the quick response! Indeed there is a corner case, in which the data contains only one pulse. I tried to write
a mirror implementation of `np.squeeze` and it seems there are something more to consider. The following is my unfinished code

```py
from numpy import AxisError

def squeeze(self, axis=None):
    if axis is None:
         slices = [0 if d == 1 else slice(None, None) for d in self.shape] 
     
     if not isinstance(axis, int):
         raise TypeError("...")

     try:
         if self.shape[axis] != 1:
             raise ValueError(cannot select an axis to squeeze out which has size not equal to one)
     except IndexError as e:
          raise AxisError(str(e))

     ...
     # take care of negative axis     

     # fix the shape and ndim
```

Comment (takluyver) [reply extra_data/stacking.py]: Oh, I'd missed that squeeze takes an axis. If I know which dim I want to drop, I tend to do it like `arr[:, 0]` . But it should be easy to add.

File: extra_data/stacking.py [hunk 2]
```diff
@@ -208,3 +208,22 @@ def asarray(self):
...
+        if axis is None:
+            slices = [0 if d == 1 else slice(None, None) for d in self.shape]
+        elif isinstance(axis, (int, tuple)):
+            if isinstance(axis, int):
+                axis = (axis,)
+
+            slices = [slice(None, None)] * self.ndim
+
+            for ax in axis:
+                if self.shape[ax] != 1:
```
Comment (zhujun98) [inline extra_data/stacking.py]: Great! LGTM except a nitpick, for out-of-bound axis this code raise `IndexError` while `np.squeeze` raises `AxisError`.

Comment (takluyver) [reply extra_data/stacking.py]: True, should be fixed now.

Comment (zhujun98) [reply extra_data/stacking.py]: Thank you!

**Pull request merged**